app/components/PersonNameField.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback } from "react";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import { useForm } from "react-hook-form";
+import { Person } from "@/types/person";
+import { useMutation } from "@/utils/graphql";
+
+interface PersonNameFieldProps {
+  /**
+   * Record corresponding to row being edited
+   */
+  record: Person;
+  /**
+   * Name of record mutation
+   */
+  mutation: string;
+  /**
+   * The function to call on cancel button click, which is expected
+   * to set the parent component's edit state
+   */
+  onCancel: () => void;
+  /**
+   * Function that is an async wrapper around data refetch
+   */
+  onSaveCallback: () => Promise<void>;
```

Comment:
greattt comments!! makes me think i should have done comments in my interface for the swap address component but it was only 3 self explanatory props, but maybe we should just make this a standard in our code to always docstring the interface props :shrug: